docs/ingestion/data-formats.md

```diff
@@ -606,11 +606,12 @@ Flattening is only supported for [data formats](data-formats.md) that support ne
+| name  | Name of the field after flattening. This name can be referred to by the [`timestampSpec`](./ingestion-spec.md#timestampspec), [`transformSpec`](./ingestion-spec.md#transformspec), [`dimensionsSpec`](./ingestion-spec.md#dimensionsspec), and [`metricsSpec`](./ingestion-spec.md#metricsspec).                                                                                                                                                                                                                                                                                                                                                                                                                                | none (required)                          |
+| expr  | Expression for accessing the field while flattening. For type `path`, this should be [JsonPath](https://github.com/jayway/JsonPath). For type `jq`, this should be [jackson-jq](https://github.com/eiiches/jackson-jq) notation. For other types, this parameter is ignored.                                                                                                                                                                                                                                                                                                                                                                                                                                                     | none (required for types `path` and `jq`) |
+| exprs | For `tree` only. Multiple-expression field for accessing the field while flattening, representing the hierarchy of field names to read. For other types, this parameter must not be provided.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    | none (required for `tree`)               |
```

Comment:
Since `avro`, `parquet`, `orc` format is not fully supported by this new feature, it's better to document these constraints here.

Reply:
Fixed in 429fd0d13a.

I've added some extras specifically in the Avro, ORC, and Parquet sections.

Also I'm not familiar with Avro, ORC, or Parquet, but would welcome any guidance on how to correctly implement this as I think the implementation could be simple enough.

Reply:
Also fixed a typo in 4f80239.